Assets/Scripts/GameplayObjects/IPooledObject.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public enum PooledObjectType
```

Comment:
this type describes every object in the game, too generic. follow the main comment at the end of this to remove it completely too